posthog/models/integration.py

```diff
@@ -301,10 +329,17 @@ def integration_from_oauth_response(
 
         if oauth_config.token_info_url:
             # If token info url is given we call it and check the integration id from there
-            token_info_res = requests.get(
-                oauth_config.token_info_url.replace(":access_token", config["access_token"]),
-                headers={"Authorization": f"Bearer {config['access_token']}"},
-            )
+            if oauth_config.token_info_graphql_query:
+                token_info_res = requests.post(
+                    oauth_config.token_info_url,
+                    headers={"Authorization": f"Bearer {config['access_token']}"},
+                    json={"query": oauth_config.token_info_graphql_query},
+                )
```

Comment:
@PostHog/team-cdp this is probably the most interesting part for ya'll. Linear only has a GraphQL API so I had to change the request type slightly to exchange the token

Reply:
the hipsters api